pootle/apps/pootle_translationproject/models.py

```diff
@@ -344,11 +323,30 @@ def is_accessible_by(self, user):
 
         return self.project.code in Project.accessible_by_user(user)
 
-    def update(self):
-        """Update all stores to reflect state on disk"""
-        stores = self.stores.live().exclude(file='').filter(state__gte=PARSED)
+    def update_from_disk(self, force=False, overwrite=False):
+        changed = False
+        # Create new, make obsolete in-DB stores to reflect state on disk
+        self.scan_files()
+
+        stores = self.stores.live().select_related('parent').exclude(file='')
+        # Update store content from disk store
         for store in stores.iterator():
-            store.update_from_disk()
+            if not store.file:
+                continue
+            disk_mtime = store.get_file_mtime()
+            if not force and disk_mtime == store.file_mtime:
+                # The file on disk wasn't changed since the last sync
+                logging.debug(u"File didn't change since last sync, "
+                              "skipping %s", store.pootle_path)
+                continue
+
+            changed = store.update_from_disk(overwrite=overwrite) or changed
+
+        # If this TP has no stores, cache should be updated forcibly.
+        if not changed and stores.count() == 0:
```

Comment:
no because if the store changed it will have triggered an update to cache already